src/client.toit

```diff
@@ -127,8 +127,8 @@ class Client:
       --uri/string
       --headers/Headers=Headers:
     parsed := parse_ uri --web_socket=false
-    connection := new_connection_ parsed --auto_close
-    request := connection.new_request method parsed.path headers
+    ensure_connection_ parsed
```

Comment:
Maybe it is silly, but I feel like `ensure_connect_` should return the connection, so you avoid using `connection_` in most places.